# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditHelpers.h
# Commit: 20512741e7b6
# Full Hash: 20512741e7b60f3f6791002d8311d681f9242e68
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-02-02 09:15:42
# Regressor Bug: 1807829
# File Overlap Count: 4
# Description:
#   Bug 1812170 - Make `HTMLEditor::ClearStyleAt` should return `pointToPutCaret` if next node of first split is not split r=m_kato
#   
#   The new path added in bug 1807829 hits the odd result case of `ClearStyleAt`.
#   `pointToPutCaret` may be updated if the preceding split occurs.  Therefore,
#   it should be returned if it does not split next nodes after the first split
# ==============================================================================

diff -r d9138ddf3c3d -r 20512741e7b6 editor/libeditor/HTMLEditHelpers.h
--- a/editor/libeditor/HTMLEditHelpers.h	Thu Feb 02 02:25:04 2023 +0000
+++ b/editor/libeditor/HTMLEditHelpers.h	Thu Feb 02 05:09:15 2023 +0000
@@ -373,7 +373,7 @@
         mDirection(aDirection) {}
 
   SplitNodeResult ToHandledResult() const {
-    mHandledCaretPoint = true;
+    CaretPointHandled();
     SplitNodeResult result(mDirection);
     result.mPreviousNode = GetPreviousContent();
     result.mNextNode = GetNextContent();
@@ -381,7 +381,7 @@
     // Don't recompute the caret position because in this case, split has not
     // occurred yet.  In the case,  the caller shouldn't need to update
     // selection.
-    result.mCaretPoint = mCaretPoint;
+    result.SetCaretPoint(CaretPointRef());
     return result;
   }
 
@@ -448,7 +448,7 @@
     aSplitNodeResult.UnmarkAsHandledCaretPoint();
     aDeeperSplitNodeResult.IgnoreCaretPointSuggestion();
     if (aSplitNodeResult.DidSplit() ||
-        !aDeeperSplitNodeResult.mCaretPoint.IsSet()) {
+        !aDeeperSplitNodeResult.HasCaretPointSuggestion()) {
       return std::move(aSplitNodeResult);
     }
     SplitNodeResult result(std::move(aSplitNodeResult));
@@ -495,12 +495,7 @@
   // for representing the point.
   EditorDOMPoint mGivenSplitPoint;
 
-  // The point which is a good point to put caret from point of view the
-  // splitter.
-  EditorDOMPoint mCaretPoint;
-
   SplitNodeDirection mDirection;
-  bool mutable mHandledCaretPoint = false;
 };
 
 /*****************************************************************************